lib/nn-syntax/tonn_converter.js

```diff
@@ -282,12 +282,37 @@ module.exports = class ToNNConverter {
             else
                 return this.findEntity('LOCATION', value);
         } else if (value.isDate) {
-            if (value.value === null)
+            if (value.value === null) {
                 return 'now';
-            else if (value.value instanceof Ast.DateEdge)
+            } else if (value.value instanceof Ast.DateEdge) {
                 return List.concat(value.value.edge, 'unit:' + value.value.unit);
-            else
+            } else if (value.value instanceof Ast.DatePiece) {
+                let toConcat = ['new', 'DatePiece', '('];
```

Comment:
`List` are designed for O(1) concatenation, so you don't need to wrangle with Arrays. You can write:
```
let stuff = List.concat('new', 'DatePiece', '(');
if (value.value.originalYear !== null) {
    stuff = List.concat(stuff, String(value.value.originalYear));
}
```
etc. etc.